fastlane_bot/events/listener.py

```diff
@@ -0,0 +1,72 @@
+import asyncio
+import time
+from typing import AsyncGenerator, List
+
+from web3 import AsyncWeb3
+
+from .utils import complex_handler
+from .managers.base import BaseManager
+from .interfaces.event import Event
+
+
+class EventListener:
+    """
+    The EventListener is the entry point to create & manage websocket event subscriptions.
+    """
+    NEW_EVENT_TIMEOUT = 0.01
+
+    def __init__(
+        self,
+        manager: BaseManager,
+        base_exchanges: List[str],
+        w3: AsyncWeb3,
+    ):
+        """ Initializes the EventManager.
```

Comment:
I tend to attack the init doc string to the class because of the call semantics

obj = MyClass(a,b,c)

as opposed to 

obj = MyClass.__init__(a,b,c)